### PR TITLE
Modifications for demo v1.0.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

**Comment by dkorolev** [comment]
Assigned @deathbaba 
CC @sompylasar @mzhurovich

File: sherlock.h [hunk 1]
```diff
@@ -84,12 +89,44 @@
...
+  inline bool Entry(E& entry) {
+    try {
+      http_response_(entry, value_name_);
+      return true;
+    } catch (const bricks::net::NetworkException&) {
+      return false;
+    }
+  }
+
+  inline void Terminate() { http_response_("{\"error\":\"Done.\"}\n"); }
```
Comment (sompylasar) [inline sherlock.h]: `s/Done./Terminated./`?

File: sherlock.h [hunk 2]
```diff
@@ -181,8 +218,28 @@ class StreamInstanceImpl {
...
+            //
+            // Thus, here we need to make a copy.
+            // The below implementation is imperfect, but it serves the purpose semantically.
+            // TODO(dkorolev): Fix it.
+
+            const std::string json = JSON(data[cursor]);
+            T copy_of_entry;
+            try {
+              ParseJSON(json, copy_of_entry);
+            } catch (const std::exception& e) {
```
Comment (biodranik) [inline sherlock.h:235]: So server can be easily killed by invalid JSON, right?
Is it a good design?

File: sherlock.h (hunk 1 above)
Comment (dkorolev) [inline sherlock.h]: Done.

File: sherlock.h (hunk 2 above)
Comment (dkorolev) [inline sherlock.h:235]: The stream can be easily killed if it be being fed with data that can not be de-serialized right after being serialized.

On the backend, preventing data corruption has priority. Yes, it's a good design.

**Comment by dkorolev** [comment]
@sompylasar @mzhurovich @deathbaba : PTAL.

**Comment by dkorolev** [comment]
@deathbaba : Status check?

I've added a commit to have the code run on Mac.

Thanks,
Dima

**Comment by dkorolev** [comment]
@deathbaba : Status check 2?

**Pull request merged**